app/src/main/java/com/keylesspalace/tusky/components/trending/TrendingFragment.kt

```diff
@@ -0,0 +1,364 @@
+                            }
+                        } else if (!composeButton.isShown) {
+                            composeButton.show()
+                        }
+                    }
+                }
+            })
+        }
+
+        viewLifecycleOwner.lifecycleScope.launch {
+            viewModel.state.collectLatest { trendingState ->
+                processViewState(trendingState)
+            }
+        }
+
+        eventHub.events
+            .observeOn(AndroidSchedulers.mainThread())
+            .autoDispose(this, Lifecycle.Event.ON_DESTROY)
+            .subscribe { event ->
+                when (event) {
+                    is PreferenceChangedEvent -> {
+                        onPreferenceChanged(event.preferenceKey)
+                    }
+                }
+            }
+    }
+
+    private fun setupSwipeRefreshLayout() {
+        binding.swipeRefreshLayout.isEnabled = isSwipeToRefreshEnabled
+        binding.swipeRefreshLayout.setOnRefreshListener(this)
+        binding.swipeRefreshLayout.setColorSchemeResources(R.color.tusky_blue)
+    }
+
+    private fun setupRecyclerView() {
+        binding.recyclerView.setHasFixedSize(true)
+        binding.recyclerView.layoutManager = LinearLayoutManager(context)
+        val divider = DividerItemDecoration(context, RecyclerView.VERTICAL)
+        binding.recyclerView.addItemDecoration(divider)
+
+        // CWs are expanded without animation, buttons animate itself, we don't need it basically
+        (binding.recyclerView.itemAnimator as SimpleItemAnimator).supportsChangeAnimations = false
+        binding.recyclerView.adapter = adapter
+    }
+
+    override fun onRefresh() {
+        viewLifecycleOwner.lifecycleScope.launch {
+            viewModel.invalidate()
+        }
+    }
+
+    override fun onViewUrl(url: String) {
+        bottomSheetActivity.viewUrl(url, PostLookupFallbackBehavior.OPEN_IN_BROWSER)
+    }
+
+    override fun onViewTag(tag: String) {
+        startActivity(StatusListActivity.newHashtagIntent(requireContext(), tag))
+    }
+
+    override fun onViewAccount(id: String) {
+        bottomSheetActivity.viewAccount(id)
+    }
+
+    private fun processViewState(trendingState: TrendingViewModel.State) {
+        when (trendingState.loadingState) {
+            TrendingViewModel.LoadingState.INITIAL -> clearLoadingState()
+            TrendingViewModel.LoadingState.LOADING -> applyLoadingState()
+            TrendingViewModel.LoadingState.LOADED -> applyLoadedState(trendingState.trendingViewData)
+            TrendingViewModel.LoadingState.ERROR_NETWORK -> networkError()
+            TrendingViewModel.LoadingState.ERROR_OTHER -> otherError()
+        }
+    }
+
+    private fun applyLoadedState(viewData: List<TrendingViewData>) {
+        clearLoadingState()
+        adapter.submitList(viewData)
+        Log.v("TESTTEST", "list: ${viewData.map { it.asTagOrNull()?.tag?.name }}")
```

Comment:
Holdover, and can be removed? If not, please use the `TAG` const.

Reply:
:+1: Resolved locally.